tests/ops/service/connectors/test_connector_template_loaders.py

```diff
@@ -43,35 +55,117 @@ def test_file_connector_template_loader(self):
-        connector_templates = loader.get_connector_templates()
+    @pytest.fixture(autouse=True)
+    def reset_custom_connector_template_data(self):
+        """
+        Resets the loader singleton instances before each test
+        """
+        FileConnectorTemplateLoader._instance = None
+        CustomConnectorTemplateLoader._instance = None
+
+    @pytest.fixture
+    def zendesk_config(self) -> str:
+        return load_yaml_as_string("data/saas/config/zendesk_config.yml")
+
+    @pytest.fixture
+    def zendesk_dataset(self) -> str:
+        return load_yaml_as_string("data/saas/dataset/zendesk_dataset.yml")
+
+    @pytest.fixture
+    def replaceable_zendesk_config(self) -> str:
+        return load_yaml_as_string(
+            "tests/fixtures/saas/test_data/replaceable_zendesk_config.yml"
+        )
+
+    @pytest.fixture
+    def replaceable_planet_express_config(self) -> str:
+        return load_yaml_as_string(
+            "tests/fixtures/saas/test_data/planet_express/replaceable_planet_express_config.yml"
+        )
+
+    @pytest.fixture
+    def replaceable_zendesk_zip(
+        self, replaceable_zendesk_config, zendesk_dataset
+    ) -> BytesIO:
+        return create_zip_file(
+            {
+                "config.yml": replace_version(replaceable_zendesk_config, "0.0.0"),
+                "dataset.yml": zendesk_dataset,
+            }
+        )
+
+    @pytest.fixture
+    def non_replaceable_zendesk_zip(self, zendesk_config, zendesk_dataset) -> BytesIO:
+        return create_zip_file(
+            {
+                "config.yml": replace_version(zendesk_config, "0.0.0"),
+                "dataset.yml": zendesk_dataset,
+            }
+        )
+
+    @pytest.fixture
+    def replaceable_planet_express_zip(
+        self,
+        replaceable_planet_express_config,
+        planet_express_dataset,
+        planet_express_functions,
+        planet_express_icon,
+    ) -> BytesIO:
+        return create_zip_file(
+            {
+                "config.yml": replaceable_planet_express_config,
+                "dataset.yml": planet_express_dataset,
+                "icon.svg": planet_express_icon,
+                "functions.py": planet_express_functions,
+            }
+        )
+
+    @pytest.fixture
+    def non_replaceable_zendesk_zip(self, zendesk_config, zendesk_dataset) -> BytesIO:
+        return create_zip_file(
+            {
+                "config.yml": replace_version(zendesk_config, "0.0.0"),
+                "dataset.yml": zendesk_dataset,
+            }
+        )
+
+    def test_custom_connector_template_loader_no_templates(self):
+        CONFIG.security.allow_custom_connector_functions = True
+
+        connector_templates = CustomConnectorTemplateLoader.get_connector_templates()
         assert connector_templates == {}
 
-    def test_file_connector_template_loader_invalid_template(
-        self, db, planet_express_dataset, planet_express_icon, planet_express_functions
+    @mock.patch(
```

Comment:
Mocked the database so these tests didn't interfere with each other, plus it's good practice right 🤷?